tools/repack-kernel

```diff
@@ -0,0 +1,235 @@
+#!/bin/bash
+set -e
+
+if [ -n "$D" ]; then
+    set -x
+fi
+
+show_help() {
+    echo "usage: repack-kernel <command> <opts>"
+    echo ""
+    echo "handle extraction of the kernel snap to a workspace directory, and later"
+    echo "repacking it back to a snap."
+    echo ""
+    echo "   repack-kernel setup                        - setup system dependencies"
+    echo "   repack-kernel extract <snap-file> <target> - extract under <target> workspace tree"
+    echo "   repack-kernel prepare <target>             - prepare initramfs & kernel for repacking"
+    echo "   repack-kernel pack <target>                - pack the kernel"
+    echo "   repack-kernel cull-firmware <target>       - remove unnecessary firmware"
+    echo "   repack-kernel cull-modules <target>        - remove unnecessary modules"
+    echo ""
+}
+
+setup() {
+    if [ "$UID" != "0" ]; then
+        echo "run as root (only this command)"
+        exit 1
+    fi
+
+    # carries ubuntu-core-initframfs
+    add-apt-repository ppa:snappy-dev/image -y
```

Comment:
I would check first if ubuntu-core-initramfs is installed, in case it is not then we can add the ppa and continue with the installation. The setup could be executed more than once in different tests.